modules/ding_place2book/ding_place2book.module

```diff
@@ -416,55 +500,50 @@ function ding_place2book_get_event_state($event_maker_id, $event_id) {
     $p2b = ding_place2book_instance();
     $event = $p2b->getEvent($event_maker_id, $event_id);
     $meta_data = $event->meta_data;
-    $event_state = $meta_data->event_state;
-    $start = strtotime($event->begin_at);
-    $end = strtotime($event->end_at);
+    $event_state = $event->state;
     $available_tickets = $meta_data->available_tickets;
-    $now = time();
 
-    // Try to derive an event status for the event by looking at the p2b event
-    // state, start date, available tickets and ticket types sales periods.
-    // See: http://developer.place2book.com/v1/events/ for a list possible p2b
-    // event states and what they mean.
+    // Try to derive a status for the ticket link in frontend by looking at the
+    // p2b event's state, ticket types and sale periods.
+    // See: http://developer.place2book.com/v1/events/
     $status = '';
-    // Start out by rulling out some easy conditions looking at the events start
-    // and end date.
-    if ($now >= $start && $now <= $end) {
-      // The event is currently taking place and ticket sales is over.
-      $status = 'closed';
-    }
-    elseif ($now > $end) {
-      // The event has already taking place.
+    if (in_array($event_state, ['held', 'completed'])) {
       $status = 'event-over';
     }
+    elseif ($event_state === 'pause') {
+      $status = 'closed';
+    }
     elseif ($event_state === 'published') {
-      // Now we know the event is not passive and is not over/taking place. If
-      // it's published it can now either be open, upcoming, closed or waiting
-      // list. Find out which by looking at the tickets and ticket types.
+      // If event is publihshed we need to look at ticket types and sale
+      // periods.
       if ($available_tickets > 0 || strval($available_tickets) == '∞') {
```

Comment:
Do we need to do this, or can we use `$meta_data->sales_status`?

Reply:
We can't it's not reliable.